modules/runner/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItDistributionZonesTest.java

```diff
@@ -0,0 +1,258 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Test suite for rebalance process.
+ */
+@ExtendWith(WorkDirectoryExtension.class)
+@Timeout(90)
+public class ItDistributionZonesTest {
+    private static final IgniteLogger LOG = Loggers.forClass(ItDistributionZonesTest.class);
+
+    /**
+     * Nodes bootstrap configuration pattern.
+     *
+     * <p>rpcInstallSnapshotTimeout is changed to 10 seconds so that sporadic snapshot installation failures still
+     * allow tests pass thanks to retries.
+     */
+    private static final String NODE_BOOTSTRAP_CFG = "{\n"
+            + "  network: {\n"
+            + "    port:{},\n"
+            + "    nodeFinder:{\n"
+            + "      netClusterNodes: [ {} ]\n"
+            + "    }\n"
+            + "  },\n"
+            + "  raft.rpcInstallSnapshotTimeout: 10000"
+            + "}";
+
+    @WorkDirectory
+    private Path workDir;
+
+    private Cluster cluster;
+
+    @BeforeEach
+    void createCluster(TestInfo testInfo) {
+        cluster = new Cluster(testInfo, workDir, NODE_BOOTSTRAP_CFG);
+    }
+
+    @AfterEach
+    @Timeout(60)
+    void shutdownCluster() {
+        cluster.shutdown();
+    }
+
+    @Test
+    @Disabled
+    void assingmentsChangingOnNodeLeaveNodeJoin() throws Exception {
+        cluster.startAndInit(4);
+
+        createTestTable();
+
+        assertTrue(waitAssingments(List.of(
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 2),
+                Set.of(0, 1, 2)
+        )));
+
+        TableImpl table = (TableImpl) cluster.node(0).tables().table("TEST");
+
+        BinaryRowEx key = new TupleMarshallerImpl(table.schemaView()).marshal(Tuple.create().set("id", 1));
+
+        assertNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(0).node()).get());
+        assertNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(1).node()).get());
+        assertNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(2).node()).get());
+
+        putData();
+
+        assertNotNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(0).node()).get());
+        assertNotNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(1).node()).get());
+        assertNotNull(table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(2).node()).get());
+
+        try {
+            table.internalTable().get(key, new HybridClockImpl().now(), cluster.node(3).node()).get();
+
+            fail();
+        } catch (Exception e) {
+            assertInstanceOf(ExecutionException.class, e);
+
+            assertInstanceOf(ReplicaUnavailableException.class, e.getCause());
+        }
+
+        cluster.knockOutNode(2, PARTITION_NETWORK);
+
+        assertTrue(waitAssingments(List.of(
+                Set.of(0, 1, 3),
+                Set.of(0, 1, 3),
+                Set.of(0, 1, 2),
```

Comment:
Why do we have `Set.of(0, 1, 2)` here?

Reply:
Because the node2 is disjointed from cluster and has old assignments.